test/integration/vaults/roles.integration.ts

```diff
@@ -0,0 +1,406 @@
+import { expect } from "chai";
+import { ContractMethodArgs, ContractTransactionReceipt, ZeroAddress } from "ethers";
+import { ethers } from "hardhat";
+import { beforeEach } from "mocha";
+
+import { HardhatEthersSigner } from "@nomicfoundation/hardhat-ethers/signers";
+
+import { Delegation } from "typechain-types";
+
+import { days } from "lib";
+import { getProtocolContext, ProtocolContext } from "lib/protocol";
+import { getRandomSigners } from "lib/protocol/helpers/get-random-signers";
+
+import { Snapshot } from "test/suite";
+
+const VAULT_OWNER_FEE = 1_00n; // 1% AUM owner fee
+const VAULT_NODE_OPERATOR_FEE = 1_00n; // 3% node operator fee
+
+type Methods<T> = {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  [K in keyof T]: T[K] extends (...args: any) => any ? K : never; // gdfg
+}[keyof T];
+
+type DelegationMethods = Methods<Delegation>; // "foo" | "bar"
+
+describe("Scenario: Staking Vaults Delegation Roles", () => {
```

Comment:
```suggestion
describe("Integration: Staking Vaults Delegation Roles Initial Setup", () => {
```
It's not like a scenario test, but more regular integration + updated title a bit